### PR TITLE
feat: add effects for note freeze

### PR DESCRIPTION
**Pull request by nmpsilva**
This pull request adds support for freezing and unfreezing notes in the SDK, including new effect types, event types, effect classes, and corresponding tests. These changes enable programmatically freezing a note (preventing modifications for a set duration) and unfreezing it, with proper payload validation and event tracking.

**SDK and Effect System Enhancements:**

* Added new effect classes `FreezeNoteEffect` and `UnfreezeNoteEffect` in `canvas_sdk.effects.note.freeze`, allowing notes to be frozen (with optional blur and user exceptions) and later unfrozen. These classes include validation and payload generation logic.
* Updated `canvas_sdk.effects.note.__init__.py` to export the new freeze/unfreeze effect classes, making them available for import elsewhere in the SDK. [[1]](diffhunk://#diff-26c3a58973aea56054991b71faa690b5e983fafc6df36558cdc25b70bc8ee187R8) [[2]](diffhunk://#diff-26c3a58973aea56054991b71faa690b5e983fafc6df36558cdc25b70bc8ee187R18-R19)
* Added comprehensive unit tests for the freeze and unfreeze note effects, covering payload structure, validation, and usage scenarios in `canvas_sdk/tests/effects/note/test_freeze.py`.

**Protobuf and Enum Updates:**

* Introduced new effect types `FREEZE_NOTE` and `UNFREEZE_NOTE` to the `EffectType` enum in both the protobuf definition (`protobufs/canvas_generated/messages/effects.proto`) and the generated Python stubs, enabling these actions to be referenced and serialized. [[1]](diffhunk://#diff-8430c8583dc0575bdf4a4c4291d6959e3d30399e27918ad504ad54c1c975a3e2R337-R339) [[2]](diffhunk://#diff-3468da6a8c62d1e9717876e5f3bfe5ce43b7c7e9d1eb68d0431d74f0d06ca94dR275-R276) [[3]](diffhunk://#diff-3468da6a8c62d1e9717876e5f3bfe5ce43b7c7e9d1eb68d0431d74f0d06ca94dR599-R600)
* Added new event types related to note lifecycle and freezing (`NOTE_CREATED`, `NOTE_BODY_UPDATED`, `NOTE_FROZEN`, `NOTE_UNFROZEN`) to the `EventType` enum in both the protobuf and generated Python stubs, supporting event tracking for these actions. [[1]](diffhunk://#diff-cae36add6ee9f014dd1a621ba2c0e9fbf0feabf183e5f87c1babd1ad6942840dR77-R85) [[2]](diffhunk://#diff-2368209f7edf8887a08fbc526d42d4dd1b9304f0e8832e668fa0b37f0d431e5dR75-R79) [[3]](diffhunk://#diff-2368209f7edf8887a08fbc526d42d4dd1b9304f0e8832e668fa0b37f0d431e5dR1139-R1143)

**Import and Plugin Runner Configuration:**

* Updated `plugin_runner/allowed-module-imports.json` to allow importing the new freeze/unfreeze effect classes and their module, ensuring plugin compatibility and proper module access. [[1]](diffhunk://#diff-a40dcade76413e693e0a9fefc6afe070adb5765f6ca17ff86b3982eef6df5c32R503-R507) [[2]](diffhunk://#diff-a40dcade76413e693e0a9fefc6afe070adb5765f6ca17ff86b3982eef6df5c32R520-R523)

[KOALA-4025]: https://canvasmedical.atlassian.net/browse/KOALA-4025?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: canvas_sdk/effects/note/freeze.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        duration: The freeze duration in seconds (default: 300, minimum: 1).
+        user_id: The staff key of the user who initiated the freeze (optional).
+                 If provided, this user will still be able to edit the note.
+        blur: Whether to blur the note content for other users (default: False).
+    """
+
+    class Meta:
+        effect_type = EffectType.FREEZE_NOTE
+
+    note_id: UUID = Field(description="The note's id")
```
Comment (jamagalhaes) [inline canvas_sdk/effects/note/freeze.py]: Can you please check if strs are accepted if we set `strict=False`?

**Review by jamagalhaes — APPROVED** [review]

**Review by jamagalhaes — APPROVED** [review]
LGTM. please add tests to validate the scenario where instance_id is None so we can assert that ValueError("Field 'instance_id' is required to freeze the note.") is raised
